### PR TITLE
feat: admin config init

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Refactors multiple admin-only fields into a new "admin" config.

## Context:

We want complete separation of options which users can change and those they can't.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/admin.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+let adminConfig: RepoAdminConfig = {};
+
+const derivedAdminOptions = ['localDir'];
+
+export function setAdminConfig(
+  config: RenovateConfig,
+  adminOptions: string[]
+): void {
+  adminConfig = {};
+  const repoAdminOptions = adminOptions.concat(derivedAdminOptions).sort();
```
Comment (viceice) [inline lib/config/admin.ts:12]: sort?

File: lib/config/admin.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  for (const option of repoAdminOptions) {
+    adminConfig[option] = config[option];
```
Comment (viceice) [inline lib/config/admin.ts:14]:
```suggestion
    adminConfig[option] = config[option];
    delete config[option];
```

Should we remove it from normal config then? So it's no longer passed through?

Comment (rarkins) [reply lib/config/admin.ts:14]: Only once we're finished :)

File: lib/config/admin.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/admin.ts:12]: Was using it for debugging / logging

Comment (viceice) [reply lib/config/admin.ts:12]: so can we remove it now?

Comment (rarkins) [reply lib/config/admin.ts:12]: I'll remove it before merge.

**Review by viceice — DISMISSED** [review]

**Pull request merged**